pkg/config/config.go

```diff
@@ -20,25 +20,28 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"time"
 
+	"github.com/onsi/gomega"
 	log "github.com/sirupsen/logrus"
 	"github.com/test-network-function/test-network-function/pkg/config/autodiscover"
 	"github.com/test-network-function/test-network-function/pkg/config/configsections"
+	"github.com/test-network-function/test-network-function/pkg/tnf"
+	"github.com/test-network-function/test-network-function/pkg/tnf/handlers/ipaddr"
+	"github.com/test-network-function/test-network-function/pkg/tnf/interactive"
+	"github.com/test-network-function/test-network-function/pkg/tnf/reel"
 	"gopkg.in/yaml.v2"
 )
 
 const (
 	configurationFilePathEnvironmentVariableKey = "TNF_CONFIGURATION_PATH"
 	defaultConfigurationFilePath                = "tnf_config.yml"
+	defaultTimeoutSeconds                       = 10
 )
 
 var (
-	// configInstance is the singleton instance of loaded config, accessed through GetConfigInstance
-	configInstance configsections.TestConfiguration
-	// loaded tracks if the config has been loaded to prevent it being reloaded.
-	loaded = false
-	// set when an intrusive test has done something that would cause Pod/Container to be recreated
-	needsRefresh = false
+	// testEnvironment is the singleton instance of `TestEnvironment`, accessed through `GetTestEnvironment`
+	testEnvironment TestEnvironment
```

Comment:
In here, by using ``TestEnvironment`` instead of ``testEnvironment``, users can actually instantiate new instances of this. 
by not doing it, we can't have function that takes the singleton as parameter, so each function need to actually call ``config.GetInstance()`` which is okay solution, but requires more work :) 
WDYT ?